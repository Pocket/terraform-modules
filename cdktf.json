{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
