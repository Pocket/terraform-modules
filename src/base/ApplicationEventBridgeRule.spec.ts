import { Testing } from 'cdktf';
import {
  ApplicationEventBridgeRule,
  ApplicationEventBridgeRuleProps,
} from './ApplicationEventBridgeRule';
import { SqsQueue } from '@cdktf/provider-aws';

const config: ApplicationEventBridgeRuleProps = {
  name: 'Test-EventBridge',
  eventPattern: {
    source: ['aws.states'],
    'detail-type': ['Step Functions Execution Status Change'],
  },
};

describe('AplicationEventBridgeRule', () => {
  it('renders an event bridge without a target', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationEventBridgeRule(stack, 'test-event-bridge-rule', config);
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an event bridge with sqs target', () => {
    const synthed = Testing.synthScope((stack) => {
      const sqs = new SqsQueue(stack, 'test-queue', {
        name: 'Test-SQS-Queue',
      });

      new ApplicationEventBridgeRule(stack, 'test-event-bridge-rule', {
        ...config,
        target: {
          dependsOn: sqs,
          arn: sqs.arn,
          targetId: 'sqs',
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an event bridge with description', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationEventBridgeRule(stack, 'test-event-bridge-rule', {
        ...config,
        description: 'Test description',
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an event bridge with event bus name', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationEventBridgeRule(stack, 'test-event-bridge-rule', {
        ...config,
        eventBusName: 'test-bus',
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an event bridge with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationEventBridgeRule(stack, 'test-event-bridge-rule', {
        ...config,
        tags: {
          my: 'tag',
          for: 'test',
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});
