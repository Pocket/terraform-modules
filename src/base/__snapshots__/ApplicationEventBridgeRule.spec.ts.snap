// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AplicationEventBridgeRule renders an event bridge with description 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_rule\\": {
      \\"test-event-bridge-rule_DA4F2E87\\": {
        \\"description\\": \\"Test description\\",
        \\"event_bus_name\\": \\"default\\",
        \\"event_pattern\\": \\"{\\\\\\"source\\\\\\":[\\\\\\"aws.states\\\\\\"],\\\\\\"detail-type\\\\\\":[\\\\\\"Step Functions Execution Status Change\\\\\\"]}\\",
        \\"name\\": \\"Test-EventBridge-Rule\\"
      }
    }
  }
}"
`;

exports[`AplicationEventBridgeRule renders an event bridge with event bus name 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_rule\\": {
      \\"test-event-bridge-rule_DA4F2E87\\": {
        \\"event_bus_name\\": \\"test-bus\\",
        \\"event_pattern\\": \\"{\\\\\\"source\\\\\\":[\\\\\\"aws.states\\\\\\"],\\\\\\"detail-type\\\\\\":[\\\\\\"Step Functions Execution Status Change\\\\\\"]}\\",
        \\"name\\": \\"Test-EventBridge-Rule\\"
      }
    }
  }
}"
`;

exports[`AplicationEventBridgeRule renders an event bridge with sqs target 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_rule\\": {
      \\"test-event-bridge-rule_DA4F2E87\\": {
        \\"event_bus_name\\": \\"default\\",
        \\"event_pattern\\": \\"{\\\\\\"source\\\\\\":[\\\\\\"aws.states\\\\\\"],\\\\\\"detail-type\\\\\\":[\\\\\\"Step Functions Execution Status Change\\\\\\"]}\\",
        \\"name\\": \\"Test-EventBridge-Rule\\"
      }
    },
    \\"aws_cloudwatch_event_target\\": {
      \\"test-event-bridge-rule_event-bridge-target_5807E31B\\": {
        \\"arn\\": \\"\${aws_sqs_queue.test-queue.arn}\\",
        \\"depends_on\\": [
          \\"aws_sqs_queue.test-queue\\",
          \\"aws_cloudwatch_event_rule.test-event-bridge-rule_DA4F2E87\\"
        ],
        \\"rule\\": \\"\${aws_cloudwatch_event_rule.test-event-bridge-rule_DA4F2E87.name}\\",
        \\"target_id\\": \\"sqs\\"
      }
    },
    \\"aws_sqs_queue\\": {
      \\"test-queue\\": {
        \\"name\\": \\"Test-SQS-Queue\\"
      }
    }
  }
}"
`;

exports[`AplicationEventBridgeRule renders an event bridge with tags 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_rule\\": {
      \\"test-event-bridge-rule_DA4F2E87\\": {
        \\"event_bus_name\\": \\"default\\",
        \\"event_pattern\\": \\"{\\\\\\"source\\\\\\":[\\\\\\"aws.states\\\\\\"],\\\\\\"detail-type\\\\\\":[\\\\\\"Step Functions Execution Status Change\\\\\\"]}\\",
        \\"name\\": \\"Test-EventBridge-Rule\\",
        \\"tags\\": {
          \\"for\\": \\"test\\",
          \\"my\\": \\"tag\\"
        }
      }
    }
  }
}"
`;

exports[`AplicationEventBridgeRule renders an event bridge without a target 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_rule\\": {
      \\"test-event-bridge-rule_DA4F2E87\\": {
        \\"event_bus_name\\": \\"default\\",
        \\"event_pattern\\": \\"{\\\\\\"source\\\\\\":[\\\\\\"aws.states\\\\\\"],\\\\\\"detail-type\\\\\\":[\\\\\\"Step Functions Execution Status Change\\\\\\"]}\\",
        \\"name\\": \\"Test-EventBridge-Rule\\"
      }
    }
  }
}"
`;
